core/src/main/scala/com/lightbend/paradox/ParadoxProcessor.scala

```diff
@@ -78,25 +82,127 @@ class ParadoxProcessor(reader: Reader = new Reader, writer: Writer = new Writer)
+        val writerContext = Writer.Context(loc, pages, reader, writer, new PagedErrorContext(errorCollector, page),
+          logger, currentMapping, "", "", groups, pageProperties)
+        val serializer = linkCapturer.serializer(writerContext)
+        page.markdown.accept(serializer)
+        validate(loc.next)
+      case None => ()
+    }
+
+    roots.foreach { root =>
+      validate(Some(root.location))
+    }
+
+    def ignore(path: String) = ignorePaths.exists(_.pattern.matcher(path).matches())
+
+    linkCapturer.allLinks.foreach {
+      case CapturedLinkWithSources(CapturedAbsoluteLink(uri), sources) if validateAbsolute && !ignore(uri.toString) =>
+        validateExternalLink(uri, reportErrorOnSources(errorCollector, sources), logger)
+      case CapturedLinkWithSources(CapturedRelativeLink(path, fragment), sources) if !ignore(path) =>
+        fullSite.get(path) match {
+          case Some(file) =>
+            fragment.foreach { f =>
+              validateFragment(path, Source.fromFile(file, "UTF-8").mkString, f, reportErrorOnSources(errorCollector, sources))
+            }
+          case None =>
+            reportErrorOnSources(errorCollector, sources)(s"Could not find path [$path] in site")
+        }
+      case _ =>
+      // Ignore
+    }
+
+    errorCollector.logErrors(logger)
+    errorCollector.errorCount
+  }
+
+  private def validateExternalLink(uri: URI, reportError: String => Unit, logger: ParadoxLogger) = {
+    logger.info(s"Validating external link: $uri")
+    val conn = uri.toURL.openConnection().asInstanceOf[HttpURLConnection]
+    conn.addRequestProperty("User-Agent", "Paradox Link Validator <https://github.com/lightbend/paradox>")
+    try {
+      if (conn.getResponseCode / 100 == 3) {
+        reportError(s"Received a ${conn.getResponseCode} redirect on external link, location redirected to is [${conn.getHeaderField("Location")}]")
+      } else if (conn.getResponseCode != 200) {
+        reportError(s"Error validating external link, status code was ${conn.getResponseCode}")
+      } else {
+        if (uri.getFragment != null) {
+          val content = Source.fromInputStream(conn.getInputStream).mkString
+          validateFragment(uri.toString, content, uri.getFragment, reportError)
+        }
+      }
+    } catch {
+      case NonFatal(e) =>
+        reportError(s"Exception occurred when validating external link: $e")
+        logger.debug(e)
+    } finally {
+      conn.disconnect()
+    }
+  }
+
+  private def reportErrorOnSources(errorContext: ErrorContext, sources: List[(File, Node)])(msg: String): Unit = {
+    sources.foreach {
+      case (file, node) => errorContext(msg, file, node)
+    }
+  }
+
+  private def validateFragment(path: String, content: String, fragment: String, reportError: String => Unit) = {
+    if (!content.contains("id=\"" + fragment + "\"")) {
```

Comment:
Going through the external link updates for Cinnamon and all fragments into the Akka docs are considered invalid. This is because Paradox is using the old school `name` attribute:

https://github.com/lightbend/paradox/blob/master/core/src/main/scala/com/lightbend/paradox/markdown/AnchorLink.scala#L31

Should this validate either `id` or `name` attributes? And we should probably update the anchor link rendering.

Reply:
Yeah I noticed that today. Yes, we should validate either `id` or `name`.

Reply:
https://github.com/lightbend/paradox/pull/363

javadocs also use `name` attributes, it just happened that I wasn't yet linking to any methods in javadocs in my project.